en, zh: add release notes for v1.2.0-beta.1 (#1154)
pingcap/docs-tidb-operator: en/releases/release-1.2.0-beta.1.md

---
title: TiDB Operator 1.2.0-beta.1 Release Notes
---

# TiDB Operator 1.2.0-beta.1 Release Notes

> 

Reviewer: ```suggestion

Release date: April 7, 2021

TiDB Operator version: 1.2.0-beta.1

```

## New Features

- Support setting customized envs for backup and restore jobs ([#3833](https://github.com/pingcap/tidb-operator/pull/3833), [@dragonly](https://github.com/dragonly))
- TidbMonitor add additional volume and volumeMount configurations. ([#3855](https://github.com/pingcap/tidb-operator/pull/3855), [@mikechengwei](https://github.com/mikechengwei))
- Support affinity and tolerations in backup/restore jobs. ([#3835](https://github.com/pingcap/tidb-operator/pull/3835), [@dragonly](https://github.com/dragonly))

## Improvements

- Fix PVC resize for TiDB. ([#3891](https://github.com/pingcap/tidb-operator/pull/3891), [@dragonly](https://github.com/dragonly))
- Fix support for multiple PVC for PD. ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))
- Fix support for multiple PVC for TiKV. ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))

> - Fix the issue that multiple PVC is treated wrong in PVC resizer. ([#3858](https://github.com/pingcap/tidb-operator/pull/3858), [@dragonly](https://github.com/dragonly))

Reviewer: This should be in `bug fixes` category.

Author: done

---
title: TiDB Operator 1.2.0-beta.1 Release Notes
---

# TiDB Operator 1.2.0-beta.1 Release Notes

Release date: April 7, 2021

TiDB Operator version: 1.2.0-beta.1

## New Features

- Support setting customized environment variables for backup and restore job containers ([#3833](https://github.com/pingcap/tidb-operator/pull/3833), [@dragonly](https://github.com/dragonly))
- TidbMonitor add additional volume and volumeMount configurations ([#3855](https://github.com/pingcap/tidb-operator/pull/3855), [@mikechengwei](https://github.com/mikechengwei))
- Support affinity and tolerations in backup/restore CR ([#3835](https://github.com/pingcap/tidb-operator/pull/3835), [@dragonly](https://github.com/dragonly))
- The resources in the tidb-operator chart use the new service account when `appendReleaseSuffix` is set to `true` ([#3819](https://github.com/pingcap/tidb-operator/pull/3819), [@DanielZhangQD](https://github.com/DanielZhangQD))

## Improvements

- Add retry for DNS lookup failure exception in TiDBInitializer ([#3884](https://github.com/pingcap/tidb-operator/pull/3884), [@handlerww](https://github.com/handlerww))

> - Optimze thanos example yaml files ([#3726](https://github.com/pingcap/tidb-operator/pull/3726), [@mikechengwei](https://github.com/mikechengwei))

Reviewer: ```suggestion
- Optimize thanos example yaml files ([#3726](https://github.com/pingcap/tidb-operator/pull/3726), [@mikechengwei](https://github.com/mikechengwei))
```

> - EndEvictLeader after TiKV Pod is recreated in RollingUpdate. ([#3724](https://github.com/pingcap/tidb-operator/pull/3724), [@handlerww](https://github.com/handlerww))

Reviewer: ```suggestion
- Delete the evict leader scheduler after TiKV Pod is recreated during the rolling update ([#3724](https://github.com/pingcap/tidb-operator/pull/3724), [@handlerww](https://github.com/handlerww))
```

> - Improve support multiple PVC for PD ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support multiple PVCs for PD during scaling and failover ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))
```

> - Improve support multiple PVC for TiKV ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support multiple PVCs for TiKV during scaling ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))
```

> - Improve support PVC resizing for TiDB ([#3891](https://github.com/pingcap/tidb-operator/pull/3891), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support PVC resizing for TiDB ([#3891](https://github.com/pingcap/tidb-operator/pull/3891), [@dragonly](https://github.com/dragonly))
```

---
title: TiDB Operator 1.2.0-beta.1 Release Notes
---

# TiDB Operator 1.2.0-beta.1 Release Notes

Release date: April 7, 2021

TiDB Operator version: 1.2.0-beta.1

## New Features

> - Support setting customized envs for backup and restore jobs ([#3833](https://github.com/pingcap/tidb-operator/pull/3833), [@dragonly](https://github.com/dragonly))

Reviewer 3: ```suggestion
- Support setting customized envs for backup and restore job containers ([#3833](https://github.com/pingcap/tidb-operator/pull/3833), [@dragonly](https://github.com/dragonly))
```

Reviewer: `envs` -> `environment variables`?

Author: fixed

> - TidbMonitor add additional volume and volumeMount configurations. ([#3855](https://github.com/pingcap/tidb-operator/pull/3855), [@mikechengwei](https://github.com/mikechengwei))

Reviewer 2: ```suggestion
- TidbMonitor add additional volume and volumeMount configurations ([#3855](https://github.com/pingcap/tidb-operator/pull/3855), [@mikechengwei](https://github.com/mikechengwei))
```

> - Support affinity and tolerations in backup/restore jobs. ([#3835](https://github.com/pingcap/tidb-operator/pull/3835), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support affinity and tolerations in backup/restore CR ([#3835](https://github.com/pingcap/tidb-operator/pull/3835), [@dragonly](https://github.com/dragonly))
```

## Improvements

> - Add retry for DNS lookup failure exception in TiDBInitializer. ([#3884](https://github.com/pingcap/tidb-operator/pull/3884), [@handlerww](https://github.com/handlerww))

Reviewer 2: ```suggestion
- Add retry for DNS lookup failure exception in TiDBInitializer ([#3884](https://github.com/pingcap/tidb-operator/pull/3884), [@handlerww](https://github.com/handlerww))
```

- The resources in the tidb-operator chart use the new service account when `appendReleaseSuffix` is set to `true` ([#3819](https://github.com/pingcap/tidb-operator/pull/3819), [@DanielZhangQD](https://github.com/DanielZhangQD))

> - Optimze thanos example yaml files. ([#3726](https://github.com/pingcap/tidb-operator/pull/3726), [@mikechengwei](https://github.com/mikechengwei))

Reviewer 2: ```suggestion
- Optimze thanos example yaml files ([#3726](https://github.com/pingcap/tidb-operator/pull/3726), [@mikechengwei](https://github.com/mikechengwei))
```

Reviewer 2: I think we can remove this.

- Delete evict leader scheduler after TiKV Pod is recreated during upgrade ([#3724](https://github.com/pingcap/tidb-operator/pull/3724), [@handlerww](https://github.com/handlerww))

> - Support multiple PVC for PD. ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support multiple PVC for PD ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))
```

> - Support multiple PVC for PD ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))
> - Support multiple PVC for TiKV ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))

Reviewer 3: ```suggestion
- Improve support multiple PVC for PD ([#3820](https://github.com/pingcap/tidb-operator/pull/3820), [@dragonly](https://github.com/dragonly))
- Improve support multiple PVC for TiKV ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))
```

> - Support multiple PVC for TiKV. ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support multiple PVC for TiKV ([#3816](https://github.com/pingcap/tidb-operator/pull/3816), [@dragonly](https://github.com/dragonly))
```

## Bug Fixes

> - Fix PVC resize for TiDB. ([#3891](https://github.com/pingcap/tidb-operator/pull/3891), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Support PVC resizing for TiDB ([#3891](https://github.com/pingcap/tidb-operator/pull/3891), [@dragonly](https://github.com/dragonly))
```

Reviewer 2: Move this to the `Improvements` section?

> - Fix the issue that multiple PVC is treated wrong in PVC resizer. ([#3858](https://github.com/pingcap/tidb-operator/pull/3858), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Fix the issue that PVCs will be set to incorrect size if multiple PVCs are configured for PD/TiKV ([#3858](https://github.com/pingcap/tidb-operator/pull/3858), [@dragonly](https://github.com/dragonly))
```

> - Fix the issue when TidbCluster's `.spec.tidb` is unset. ([#3852](https://github.com/pingcap/tidb-operator/pull/3852), [@dragonly](https://github.com/dragonly))

Reviewer 2: ```suggestion
- Fix the panic issue when `.spec.tidb` is not set in the TidbCluster CR with TLS enabled ([#3852](https://github.com/pingcap/tidb-operator/pull/3852), [@dragonly](https://github.com/dragonly))
```

> - Fix `UnjoinedMembers` in PD and DM cluster status. ([#3836](https://github.com/pingcap/tidb-operator/pull/3836), [@dragonly](https://github.com/dragonly))

Reviewer 2: Remove this.

> - Fix tidbmonitor externallabels unrecognized environment variables bug. ([#3785](https://github.com/pingcap/tidb-operator/pull/3785), [@mikechengwei](https://github.com/mikechengwei))

Reviewer 2: ```suggestion
- Fix the issue that some unrecognized environment variables are included in the external labels of the TidbMonitor ([#3785](https://github.com/pingcap/tidb-operator/pull/3785), [@mikechengwei](https://github.com/mikechengwei))
```
